packages/template-parser/src/index.ts

```diff
@@ -246,7 +246,7 @@ function parseForESLint(code: string, options: { filePath: string }) {
   };
 }
 
-module.exports = {
+export default {
```

Comment:
@JamesHenry 

Why not use `export { parseForESLint, parse }`?

So that it will also work with cjs + `@ts-check`?

![image](https://user-images.githubusercontent.com/8336744/125425370-e8fe6ab0-6148-477c-bcfc-35c1fdd9680e.png)

Reply:
It's written in the comment on this PR:

```
/**
 * This usage seems a little wonky, but don't want to run any risk of breaking changes
 * to the compiled output of the template-parser node module, so leaving until v13.
 */
```

Reply:
This is not blocking anything AFAICT, you can achieve everything you need to via the default export and it will work with ts-check

Reply:
It doesn't, `require('@angular-eslint/template-parser')`

```js
// index.js
const _ = require('@angular-eslint/template-parser')

console.log(_.default)
console.log(_.parseForESLint)
```

```log
$ tsc -p jsconfig.json
index.js:11:15 - error TS2339: Property 'parseForESLint' does not exist on type 'typeof import("/workspace/angular-eslint-template-parser/node_modules/@angular-eslint/template-parser/dist/index")'.

11 console.log(_.parseForESLint)
```

```log
// run `node index.js`
undefined
[Function: u]
```

Reply:
> It's written in the comment on this PR:

The current solution is actually a breaking change for typings.

Why `export { parseForESLint }` would break too? I think it will be just fine.

Reply:
@JamesHenry Hi, any possible to fix?